### PR TITLE
osd: adding PerfCounters for backoff throttle

### PR DESCRIPTION
**Pull request by chuanhong-wang**
Signed-off-by: Chuanhong wang<chuanhong.wang@163.com>

### CODE REVIEW

**Comment by chuanhong-wang** [comment]
@xiexingguo please help me review code

File: src/common/Throttle.cc [hunk 1]
```diff
@@ -251,6 +251,57 @@ void Throttle::reset()
   }
 }
 
+enum {
+  l_backoff_throttle_first = l_throttle_last+1,
```
Comment (xiexingguo) [inline src/common/Throttle.cc]: space before and after "+"

File: src/os/filestore/FileStore.cc [hunk 2]
```diff
@@ -539,8 +539,8 @@ FileStore::FileStore(CephContext* cct, const std::string &base,
   next_osr_id(0),
   m_disable_wbthrottle(cct->_conf->filestore_odsync_write ||
                       !cct->_conf->filestore_wbthrottle_enable),
-  throttle_ops(cct->_conf->filestore_caller_concurrency),
-  throttle_bytes(cct->_conf->filestore_caller_concurrency),
+  throttle_ops(g_ceph_context, "filestore_ops", cct->_conf->filestore_caller_concurrency),
```
Comment (xiexingguo) [inline src/os/filestore/FileStore.cc]: Can we use cct here instead?

File: src/os/filestore/FileStore.cc [hunk 3]
```diff
@@ -539,8 +539,8 @@ FileStore::FileStore(CephContext* cct, const std::string &base,
...
+  throttle_bytes(g_ceph_context, "filestore_bytes", cct->_conf->filestore_caller_concurrency),
```
Comment (xiexingguo) [inline src/os/filestore/FileStore.cc]: Same

**Comment by xiexingguo** [comment]
Looks good except some nits

File: src/os/filestore/FileStore.cc (hunk 3 above)
Comment (chuanhong-wang) [inline src/os/filestore/FileStore.cc]: @xiexingguo done, please help me review again

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-02-10_21:16:27-rados-wip-yuri-testing2_2017_2_11-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2017-02-11_17:35:17-rados-wip-yuri-testing2_2017_2_11---basic-smithi/

**Pull request merged**